src/Storages/Statistics/ConditionSelectivityEstimator.cpp

```diff
@@ -14,6 +15,34 @@ void ConditionSelectivityEstimator::ColumnSelectivityEstimator::merge(String par
     if (part_statistics.contains(part_name))
         throw Exception(ErrorCodes::LOGICAL_ERROR, "part {} has been added in column {}", part_name, stats->columnName());
     part_statistics[part_name] = stats;
+    stats->setTargetType(statistics_target_types);
+}
+
+bool ConditionSelectivityEstimator::ColumnSelectivityEstimator::convertToTargetType(Field & val, const String & op) const
+{
+    auto tdigest_type_it = statistics_target_types.find(StatisticsType::TDigest);
```

Comment:
The implementation of `convertToTargetType` makes some assumptions how which statistics types are able to estimate the operator `op`. The logic is similar to `ColumnStatistics::estimateLess` and `ColumnStatistics::estimateEqual`. This creates an implicit dependency and duplication where we ideally have this code in one single code.